src/main/java/pwr/isa/klama/rentalItem/RentalItemService.java

```diff
@@ -0,0 +1,327 @@
+package pwr.isa.klama.rentalItem;
+
+import jakarta.transaction.Transactional;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import pwr.isa.klama.exceptions.ResourceNotFoundException;
+import pwr.isa.klama.rentalItem.rent.*;
+import pwr.isa.klama.user.User;
+import pwr.isa.klama.user.UserRepository;
+
+import java.sql.Timestamp;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class RentalItemService {
+
+    private final RentalItemRepository rentalItemRepository;
+    private final UserRepository userRepository;
+    private final RentRepository rentRepository;
+
+    @Autowired
+    public RentalItemService(RentalItemRepository rentalItemRepository, UserRepository userRepository, RentRepository rentRepository) {
+        this.rentalItemRepository = rentalItemRepository;
+        this.userRepository = userRepository;
+        this.rentRepository = rentRepository;
+    }
+
+    public List<RentalItem> getRentalItemsForAdmin() {return rentalItemRepository.findAll(); }
+
+    public List<RentalItemDTO> getRentalItems() {
+        return rentalItemRepository.findAll().stream()
+                .filter(rentalItem -> rentalItem.getStatus() == ItemStatus.ACTIVE)
+                .map(rentalItem -> new RentalItemDTO(
+                        rentalItem.getId(),
+                        rentalItem.getName(),
+                        rentalItem.getDescription(),
+                        rentalItem.getPrice(),
+                        rentalItem.getQuantity()
+                ))
+                .collect(Collectors.toList());
+    }
+
+    public RentalItem getRentalItemById(Long rentalItemId) {
+        boolean exists = rentalItemRepository.existsById(rentalItemId);
+        if( !exists ) {
+            throw new IllegalArgumentException("Przedmiot o id " + rentalItemId + " nie istnieje");
```

Comment:
![image](https://github.com/user-attachments/assets/7fdc93ae-908a-41b3-9a5b-37e5acf48eb0)


1. Nie `IlleagalArgument` tylko, `IllegalState` - obsługujemy drugi błąd, który zwraca `401`, a nie `500`
2. To chyba podlega pod `ResourceNotFoundException` :D